### PR TITLE
Fixed tests leaking editor instance / DOM element.

### PR DESCRIPTION
**Pull request by mlewand**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Tests: Fixed tests leaking editor instances / DOM elements. See ckeditor/ckeditor5#6002.

---

### Additional information

⚠ This PR requires [ckeditor/ckeditor5-core#207](https://github.com/ckeditor/ckeditor5-core/pull/207) to pass properly ⚠

For a complete list of PRs see https://github.com/ckeditor/ckeditor5/issues/6002#issuecomment-567834563.

### CODE REVIEW

File: tests/ballooneditor.js [hunk 1]
```diff
@@ -205,8 +205,18 @@ describe( 'BalloonEditor', () => {
...
+				} )
 				.then( done )
 				.catch( done );
+
+			function removeEditorDom() {
+				// Remove DOM leftovers to not affect other tests (#6002, #6018).
+				for ( const editorBody of document.body.querySelectorAll( 'div.ck.ck-body' ) ) {
+					editorBody.remove();
+				}
+			}
```
Comment (pomek) [inline tests/ballooneditor.js]: This code affects also other tests that do not clean after execution. I am sure somebody won't understand what's going on when he needs to execute tests without this file.

Comment (pomek) [reply tests/ballooneditor.js]: But as I can see, in other PRs you did the same thing. Do I miss anything?

Comment (mlewand) [reply tests/ballooneditor.js]: I hear your point, I also had doubt about it. But then I thought that it is uncommon in our codebase to see that someone relies on editor DOM being preserved between two tests. If anything it has been a sign of a problem so far.

Then it happens just in this particular test in each build, due to unusual nature of this test case.

Finding another solution would take considerable amount of time, so I decided to go with easier solution.

I did last fixes batch of repos today and decided to extract this function to [core package](https://github.com/ckeditor/ckeditor5-core/pull/207), as it is reused also by watchdog (which also breaks editor, a lot).

**Review by pomek — APPROVED** [review]
LGTM.

**Pull request merged**